### PR TITLE
OPENTOK-51006 - Captions API support 

### PR DESCRIPTION
**Pull request by behei-vonage**
This PR adds a sample of how to use captions API in the Node environment.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,9 @@
     "prettier": "2.8.1",
     "prettier-eslint": "15.0.1",
     "serve-favicon": "~2.5.0",
-    "uglify-js": "^3.13.4"
+    "uglify-js": "^3.13.4",
```
Comment (cpettet) [inline package.json:28]: Nit. Could an `npm install` put this in alphabetical order??

File: routes/index.js [hunk 2]
```diff
@@ -97,6 +114,69 @@ router.get('/room/:name', function (req, res) {
...
+  }
+
+  res.send(captionResponse.data.captionsId);
+});
+
+/**
+ * POST /captions/stop
+ */
+router.post('/captions/stop', postBodyParser, async function (req, res) {
+  const captionsId = req.body.captionId;
```
Comment (cpettet) [inline routes/index.js:155]: Nit, `captionId` vs `captionsId` 😅

**Review by cpettet — APPROVED** [review]
LGTM Great job! Added a few comments, but not a blocker.

File: routes/index.js (hunk 2 above)
Comment (behei-vonage) [reply routes/index.js:155]: I already have a higher-scope `captionId`, don't want to confuse people with that one 😓

File: package.json (hunk 1 above)
Comment (behei-vonage) [reply package.json:28]: Doesn't look like it's doing that 😞

Comment (manchuck) [reply package.json:28]: FYI you can run `npx sort-package-json` to sort package.json. I have this as part of the `@vonage/server-sdk`. It helps curve merge conflicts

File: routes/index.js [hunk 3]
```diff
@@ -97,6 +114,69 @@ router.get('/room/:name', function (req, res) {
...
+      },
+    });
+  } catch (err) {
+    console.warn(err);
+    res.status(500);
+    res.send(`Error stopping transcription services: ${err}`);
+    return;
+  }
+
+  res.sendStatus(captionResponse.status);
```
Comment (manchuck) [inline routes/index.js:177]:
```javascript 
  try {
    const captionResponse = await axios.post(captionURL, {}, {
      headers: {
        'X-OPENTOK-AUTH': projectJWT,
        'Content-Type': 'application/json',
      },
    });
    res.sendStatus(captionResponse.status);
  } catch (err) {
    console.warn(err);
    res.status(500);
    res.send(`Error stopping transcription services: ${err}`);
  }
```

This way captionResponse does not have to be in the defined higher which could cause issues with multiple requests coming in at the same time.

File: routes/index.js [hunk 4]
```diff
@@ -97,6 +114,69 @@ router.get('/room/:name', function (req, res) {
...
+      },
+    });
+  } catch (err) {
+    console.warn(err);
+    res.status(500);
+    res.send(`Error starting transcription services: ${err}`);
+    return;
+  }
+
+  res.send(captionResponse.data.captionsId);
```
Comment (manchuck) [inline routes/index.js:148]:
```js
  try {
    const captionResponse = await axios.post(captionURL, captionPostBody, {
      headers: {
        'X-OPENTOK-AUTH': projectJWT,
        'Content-Type': 'application/json',
      },
    });
   res.send(captionResponse.data.captionsId);
  } catch (err) {
    console.warn(err);
    res.status(500);
    res.send(`Error starting transcription services: ${err}`);
  }
```

Avoid using higher-scoped variables, as multiple requests at the same time could muck up the response

File: routes/index.js [hunk 5]
```diff
@@ -61,6 +71,10 @@ router.get('/room/:name', function (req, res) {
   if (roomToSessionIdDictionary[roomName]) {
     sessionId = roomToSessionIdDictionary[roomName];
 
+    const tokenOptions = {};
+    // we need caption to be moderator role for captions to work
+    tokenOptions.role = "moderator";
+
```
Comment (manchuck) [inline routes/index.js:77]: This can be moved up outside the if block. The `tokenOptions` is set the same in both cases

File: routes/index.js [hunk 6]
```diff
@@ -2,11 +2,19 @@ const express = require('express');
 // eslint-disable-next-line new-cap
 const router = express.Router();
 const path = require('path');
+const axios = require('axios');
```
Comment (manchuck) [inline routes/index.js:5]: We prefer to use the `node-fetch` package. This way it is more in sync with the browsers `fetch` API

**Review by manchuck — CHANGES_REQUESTED** [review]

File: routes/index.js (hunk 6 above)
Comment (manchuck) [reply routes/index.js:5]: This won't be a blocker just an FYI

**Comment by behei-vonage** [comment]
Thanks for all the feedback, @manchuck! I believe I addressed all of it. Can you please review it again when you have a chance?

File: routes/index.js (hunk 6 above)
Comment (manchuck) [reply routes/index.js:5]: Yea that is fine just letting you know going forward to try and use the `node-fetch`

**Review by manchuck — APPROVED** [review]

**Pull request merged**